File: src/main/java/uk/edu/populationfitness/models/genes/cache/OnHeapGeneValues.java
package uk.edu.populationfitness.models.genes.cache;

import uk.edu.populationfitness.models.genes.GenesIdentifier;

import java.util.Collection;

/**
 * Emulates a GeneValues by creating a self identified bitset, which retains a reference to the bitset, allowing
 * the bitset to be retrieved given the identifier. This means that all bitsets created by the cache will live
 * on the heap
 */
public class OnHeapGeneValues implements GeneValues {
    @Override
    public GenesIdentifier add(long[] genesIntegers) {
        return new SelfIdentifiedGenes(genesIntegers);
    }

    @Override
    public long[] get(GenesIdentifier identifier) {
        return ((SelfIdentifiedGenes)identifier).integerGenes;
    }

    @Override
    public void retainOnly(Collection<GenesIdentifier> genesIdentifiers) {

    }

    @Override
    public void close() {

    }
}

File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/Cache/OnHeapGeneValues.cs
using System.Collections.Generic;

namespace PopulationFitness.Models.Genes.Cache
{

    /**
     * Emulates a GeneValues by creating a self identified bitset, which retains a reference to the bitset, allowing
     * the bitset to be retrieved given the identifier. This means that all bitsets created by the cache will live
     * on the heap
     */
    public class OnHeapGeneValues : IGeneValues
    {
        public IGenesIdentifier Add(long[] genesIntegers)
        {
            return new SelfIdentifiedGenes(genesIntegers);
        }

        public long[] Get(IGenesIdentifier identifier)
        {
            return ((SelfIdentifiedGenes)identifier).integerGenes;
        }

        public void RetainOnly(ICollection<IGenesIdentifier> genesIdentifiers)
        {

        }

        public void Close()
        {

        }

        public bool IsFlushable
        {
            get
            {
                return false;
            }
        }
    }
}
